src/expression.h

```diff
@@ -1160,7 +1160,9 @@ class PreExp : public UnaExp
 class AssignExp : public BinExp
 {
 public:
-    int ismemset;       // !=0 if setting the contents of an array
+    // &1 != 0 if setting the contents of an array
+    // &2 != 0 if setting the content of ref variable
+    int ismemset;
```

Comment:
Why not make it a flag right away? Code like `!(ismemset & 1)` is literally unreadable.